packages/liveblocks-react/src/comments/CommentsRoom.tsx

```diff
@@ -384,7 +452,36 @@ export function createCommentsRoom<TThreadMetadata extends BaseMetadata>(
       };
     });
 
-    const cache = useSyncExternalStoreWithSelector(
+    const cache = _useThreads(room, options);
+
+    if (cache.error) {
+      throw cache.error;
+    }
+
+    if (cache.isLoading || !cache.threads) {
+      throw revalidateCache({
+        shouldDedupe: true,
+      });
+    }
+
+    return {
+      isLoading: false,
+      threads: cache.threads,
+      error: cache.error,
+    };
+  }
+
+  function _useThreads(
+    room: Room<JsonObject, LsonObject, BaseUserMeta, Json>,
+    options: GetThreadsOptions<TThreadMetadata>
```

Comment:
Should this use `UseThreadsOptions` instead?

Reply:
Yes, better to keep it `UseThreadsOptions`. Thanks for catching this